### PR TITLE
Fix python module invocation

### PR DESCRIPTION
**Pull request by trottomv**
This changes fix the #2520 Issue

### CODE REVIEW

File: python/ruff/__main__.py [hunk 1]
```diff
@@ -3,6 +3,23 @@
 import sysconfig
 from pathlib import Path
 
+RUFF_PATHS = [
+    Path(sysconfig.get_config_var("userbase")) / "bin" / "ruff",
```
Comment (andersk) [inline python/ruff/__main__.py:7]: I expect this is wrong on Windows. The right way to get the `--user` scripts path is `sysconfig.get_path("scripts", scheme=sysconfig.get_preferred_scheme("user"))`—but that requires Python ≥ 3.10, so we’ll need to figure out the right translation for older Python.

File: python/ruff/__main__.py [hunk 2]
```diff
@@ -3,6 +3,23 @@
...
+    Path(sysconfig.get_path("scripts")) / "ruff",
```
Comment (andersk) [inline python/ruff/__main__.py:8]: We shouldn’t prefer a version from `pip install --user` *over* a currently active virtualenv.

File: python/ruff/__main__.py [hunk 3]
```diff
@@ -3,6 +3,23 @@
...
+            return ruff_path
+    raise FileNotFoundError(ruff_path)
+
+
 if __name__ == "__main__":
-    ruff = Path(sysconfig.get_path("scripts")) / "ruff"
+    try:
+        ruff = find_ruff_bin()
+    except FileNotFoundError as e:
+        raise FileNotFoundError(e) from e
```
Comment (andersk) [inline python/ruff/__main__.py:24]: This ends up raising `FileNotFoundError(FileNotFoundError(ruff_path))`. Why catch and rewrap the exception at all?

File: python/ruff/__main__.py (hunk 1 above)
Comment (charliermarsh) [reply python/ruff/__main__.py:7]: Can you think of any other projects that've had to solve this problem?

File: python/ruff/__main__.py [hunk 4]
```diff
@@ -3,6 +3,23 @@
...
+RUFF_PATHS = [
+    Path(sysconfig.get_config_var("userbase")) / "bin" / "ruff",
+    Path(sysconfig.get_path("scripts")) / "ruff",
+]
+
+
+def find_ruff_bin() -> Path:
+    """Return the ruff binary path."""
+    for ruff_path in RUFF_PATHS:
+        if ruff_path.is_file():
```
Comment (andersk) [inline python/ruff/__main__.py:15]: This doesn’t work at all on Windows, where the binary has a `.exe` extension (which had been automatically added by `os.spawnv`).

**Review by andersk — COMMENTED** [review]
Opened #2574 to fix all these issues.

**Pull request merged**